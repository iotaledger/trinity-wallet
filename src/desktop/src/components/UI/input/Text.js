import React from 'react';
import PropTypes from 'prop-types';
import css from './Input.css';

export default class TextInput extends React.PureComponent {
    static propTypes = {
        value: PropTypes.string.isRequired,
        label: PropTypes.string,
        onChange: PropTypes.func.isRequired,
    };

    render() {
        const { value, label, onChange } = this.props;

        return (
            <div className={css.input}>
                <fieldset>
                    <input type="text" value={value} onChange={(e) => onChange(e.target.value)} />
                    <small>{label}</small>
                </fieldset>
            </div>
        );
    }
}
