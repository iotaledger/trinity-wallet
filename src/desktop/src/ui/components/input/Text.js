import React from 'react';
import PropTypes from 'prop-types';
import css from './input.scss';

/**
 * Single line input component
 */
export default class Text extends React.PureComponent {
    static propTypes = {
        /** Current input value */
        value: PropTypes.string.isRequired,
        /** Should input focus when changed to true */
        focus: PropTypes.bool,
        /** Input label */
        label: PropTypes.string,
        /** Address change event function
         * @param {string} value - Current input value
         */
        onChange: PropTypes.func.isRequired,
    };

    componentDidMount() {
        if (this.props.focus) {
            this.input.focus();
        }
    }

    componentWillReceiveProps(nextProps) {
        if (!this.props.focus && nextProps.focus) {
            this.input.focus();
        }
    }

    render() {
        const { value, label, onChange } = this.props;

        return (
            <div className={css.input}>
                <fieldset>
                    <input
                        ref={(input) => {
                            this.input = input;
                        }}
                        type="text"
                        value={value}
                        onChange={(e) => onChange(e.target.value)}
                    />
                    <small>{label}</small>
                </fieldset>
            </div>
        );
    }
}
