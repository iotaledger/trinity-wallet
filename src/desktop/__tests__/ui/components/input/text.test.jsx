import React from 'react';
import { shallow } from 'enzyme';

import Text from 'ui/components/input/Text';

const props = {
    value: 'Foo Bar',
    label: 'Foo',
    onChange: jest.fn(),
};

describe('Text component', () => {
    test('Render the component', () => {
        const wrapper = shallow(<Text {...props} />);

        expect(wrapper).toMatchSnapshot();
    });

    test('Input value', () => {
        const wrapper = shallow(<Text {...props} />);

        expect(wrapper.find('input').props().value).toEqual('Foo Bar');
    });

    test('Input change callback', () => {
        const wrapper = shallow(<Text {...props} />);

        wrapper.find('input').simulate('change', { target: { value: 'Bar' } });
        expect(props.onChange).toHaveBeenLastCalledWith('Bar');
    });

    test('Input label', () => {
        const wrapper = shallow(<Text {...props} />);

        expect(wrapper.find('small').text()).toEqual('Foo');
    });

    test('Input disabled state', () => {
        const mockProps = Object.assign({}, props, { disabled: true });
        const wrapper = shallow(<Text {...mockProps} />);

        expect(wrapper.hasClass('disabled')).toBeTruthy();
    });

    test('Input placeholder', () => {
        const mockProps = Object.assign({}, props, { placeholder: 'Bar Foo Placeholder' });
        const wrapper = shallow(<Text {...mockProps} />);

        expect(wrapper.find('input').props().placeholder).toEqual('Bar Foo Placeholder')
    })
});
