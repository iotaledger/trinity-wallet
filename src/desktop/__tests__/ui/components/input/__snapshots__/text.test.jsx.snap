// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text component Render the component 1`] = `
<div
  className="input"
>
  <fieldset>
    <input
      onChange={[Function]}
      type="text"
      value="Foo Bar"
    />
    <small>
      Foo
    </small>
  </fieldset>
</div>
`;
